### ISSUE TITLE
DevCPU closing after extended use on a debian wheezy 32 bits (x86) with wmii.

### ISSUE THREAD

**Issue by L3nn0x** [HerobrinesArmy/DevCPU#33]
After being opened for a while, DevCPU will close with the message "seg fault". I had it only twice so far and I haven't found out what was causing it yet.
I'm running a debian wheezy 32 bits (x86) with wmii.

As for precisions, it had been running for an hour or two when it occurred. I was starting/stopping the emulator quite a lot and saving/modifying files (cause we don't have a debugguer yet).

**Comment by rgarlor** [comment]
After how long? Yesterday I had it around 5 hours opened (on Windows) and didn't close until I closed it myself.

**Comment by L3nn0x** [comment]
Hum, I guess it was after one or two hours. But I kept starting and stopping the dcpu emulator and saving/modifying files. (I'm on linux.)

**Comment by dilbertfan** [comment]
Please state your platform in these types of reports! D:

**Comment by L3nn0x** [comment]
Sorry, I forgot. It's better now.

**Comment by dilbertfan** [comment]
Thanks!

**Comment by L3nn0x** [comment]
Your welcome. :)

**Comment by HerobrinesArmy** [comment]
I'll keep an eye out for similar reports. My guess is that this will prove to be a Linux-only issue. How long had you had it open when it happened?

**Comment by L3nn0x** [comment]
I think it was between one and two hours, but I kept starting/stopping the emulator and saving/modifying files.

**Comment by L3nn0x** [comment]
I may have a better idea of why it's closing : It occurs when I change virtual desks and the program is in full screen. I don't know why yet, though.

**Comment by HerobrinesArmy** [comment]
0_o That's not an easy one to debug, particularly from reports alone.

**Comment by L3nn0x** [comment]
So, I got what was written in the shell, but I'm not sure it'll be of any help.
http://pastebin.com/1z9nmRTB

**Comment by HerobrinesArmy** [comment]
Well, it's probably not very helpful for the segfault, but I'm more interested in these expression evaluation errors you have in this log. I haven't seen any of those in a very long time, and I'm kind of surprised there's still a way to generate them. Is the source code you were assembling something I can take a look at?

**Comment by L3nn0x** [comment]
Yeah, of course. It's FrOSt a french OS (a pretty big project) : https://github.com/FrOSt-Foundation/FrOSt
The building is working, as I can run it without any problem.

Otherwise, I made some experiments : it seems like it's just whenever it wants to crash. I tried to make it crash at every step. No luck, it didn't. So I don't know what to think anymore. I'll try updating as updates come and keep you informed.
Edit : I don't know if it's related, but it occurs only after starting and stopping the dcpu (with the screen, the keyboard and the registers view linked) and changing virtual desktop.

**Comment by HerobrinesArmy** [comment]
![:hmm:](http://herobrinesarmy.com/smileys/_hmm__by_upsguy1997-d4rg457.gif)

**Comment by L3nn0x** [comment]
Yeah, I don't really know what to think... I'll try some more experiments.

**Comment by HerobrinesArmy** [comment]
Nice OS. :+1: 
Here's a little tip. I noticed you had a line written

<pre>   :console_loop
        set C, 0x3e ; 0x3e : ">"</pre>

I saw support for character literals in another assembler, and I thought that was neat so I made sure to support it in DevCPU. If you have a single character you want to use as a value, you can use it with 's around it. So you have the option to write it as <pre>     set C, '>'</pre> if you want.

I don't know what other assemblers support characters literals (I know at least one does, though I can't remember which), and I don't know which assemblers you want FrOSt to work with, but I thought I'd take this opportunity to mention that feature.

One of these days, I'll get around to actually documenting DevCPU, but for now, I give information out in spurts of tips. =P

**Comment by Yamakaky** [comment]
We moved to DevCPU 2 weeks ago so all the code isn't migrated (defines, literals char…) ;)

**Comment by HerobrinesArmy** [comment]
While I'm dishing out tips, here's a big one that might not be obvious to people who haven't used Eclipse before. I think the DASM code editor starts without line numbers by default. If you want to see line numbers in the code editor, you can right-click on the left edge of the editor (there's a little gray bar along the side) and select "Show Line Numbers".

If you want to try out another neat feature, here's one provided by the EGit plugin. If a file is part of a git repository, you can right-click the left side of the editor and select "Show Annotations" (or perhaps you have to right-click the file and select Team > Show Annotations, I'm not sure) and then right click the side of the editor again and select "Revisions > Combined Coloring" and "Revisions > Show Author" for a rather colorful, interesting, and often quite useful view of which commit contributed each line in the file. Note: I've seen that sometimes it doesn't seem to do much, and I don't know why that is. I didn't make that feature so I can neither take credit for it nor take bug reports for it. =P It might have to do with some properties of the repository.

**Comment by Yamakaky** [comment]
No. `git blame <file>`. XD

**Comment by L3nn0x** [comment]
Cool. Thanks for the tips. :)

I'm not sure it's Debian, I'd think more about wmii (the window manager). I'll try with another one to see if it's the same.

**Comment by HerobrinesArmy** [comment]
@Yamakaky The feature I described is EGit's take on git-blame. If you like using git-blame, then you'll probably love EGit's blame annotations. They're much more useful than what git-blame gives you.
![blame annotations](http://wiki.eclipse.org/images/2/2f/Egit-1.0-blame-ruler.png)

**Comment by Yamakaky** [comment]
No GUI ! git + tig + vim (not found good plugin for emacs…) XD

**Comment by HerobrinesArmy** [comment]
@FaerieFrOSt It's unfortunate that it's such a non-specific problem. "segfault" isn't a particularly helpful error message to get. =P
Hopefully, it's just a consequence of the hacked-together nature of the preview release. If that's the case, it might resolve itself as I rework various things to be done the more "proper" way.

**Comment by HerobrinesArmy** [comment]
@Yamakaky You're welcome to slow yourself down if you'd like. Personally, I prefer to do things that increase my productivity and ease of development rather than sticking to some sort of religious obsession with pure-text environments. xD

**Comment by Yamakaky** [comment]
^^

**Comment by L3nn0x** [comment]
I'm with the pure-text environment too, but it's only to increase my productivity... :p 
@HerobrinesArmy : Yeah, I know it's not really helpful... I'll try more stuff when I have time.

**Comment by Yamakaky** [comment]
trying not to argue, not to argue, not to -t-r-o-l-l- argue…

**Comment by HerobrinesArmy** [comment]
I'm only kidding. Quite a few members of our dev team insist on pure-text environments.
...and Linux... -_-

**Comment by Yamakaky** [comment]
We will convert you. XD

I'm not really pure-text or pure-gui, it depends a lot what I must do and what soft is open.

**Comment by L3nn0x** [comment]
Yeah, me too, but as my pc is kinda old, pure-text is great. In a single command I can copy from one end of my hard drive to a flash drive. And remove stuff recursively at the same time.

(No I'm not trying to convert you... :p )

**Comment by L3nn0x** [comment]
OKay so I tried with another window manager and I don't get this error at all. So I guess it's a wmii related problem...

**Comment by HerobrinesArmy** [comment]
Thanks for testing that. That helps a lot. :D 
I may put a notice about that on those versions.

**Comment by L3nn0x** [comment]
Your welcome, I guess you don't have to solve it now... :p 
I only tried with gnome3, I'll try with some more window managers and report.
